packages/fxa-settings/src/models/integrations/data/data.ts

```diff
@@ -0,0 +1,357 @@
+
+  @IsOptional()
+  @IsString()
+  @bind(T.snakeCase)
+  utmSource: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind(T.snakeCase)
+  utmTerm: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind(T.snakeCase)
+  flowId: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind(T.snakeCase)
+  flowBeginTime: string | undefined;
+}
+
+/**
+ * Represents data specific to integrations that support syncing.
+ */
+export class SyncBasicIntegrationData extends IntegrationData {
+  // TODO - Validation - Will @IsISO31661Alpha2() work?
+  @IsOptional()
+  @IsString()
+  @MinLength(2)
+  @MaxLength(7)
+  @bind()
+  country: string | undefined;
+
+  @IsOptional()
+  @IsBase64()
+  @Length(8)
+  @bind()
+  signinCode: string | undefined;
+
+  // TODO - Validation - Double check actions
+  @IsOptional()
+  @IsIn(['signin', 'signup', 'email', 'force_auth', 'pairing'])
+  @bind()
+  action: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  syncPreference: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  multiService: boolean | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  tokenCode: string | undefined;
+}
+
+/**s
+ * Represents data specific general 'web' integrations, which are typically login processes where a user
+ * is trying to authenticating in oder to access their account settings page.
+ */
+export class WebIntegrationData extends IntegrationData {
+  @IsOptional()
+  // TODO - Validation - change this to 'IsEmail'. Requiring this to be an email
+  // causes this to throw when we have a separate model, like a query param or
+  // link model for a page, where we validate this instead.
+  @IsString()
+  @bind()
+  email: string | undefined;
+
+  @IsOptional()
+  @IsEmail()
+  @bind()
+  emailToHashWith: string | undefined;
+
+  @IsOptional()
+  @IsIn(['true', 'false', true, false])
+  @bind(T.snakeCase)
+  resetPasswordConfirm: boolean | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  setting: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  style: string | undefined;
+
+  @IsOptional()
+  @IsHexadecimal()
+  @Length(32)
+  @bind()
+  uid: string | undefined;
+}
+
+/**
+ * Represents data required for logging with OAuth. This is typical when a user directed to FxA from an relying party
+ * and the relying party is attempting to FxA us to validate the user.
+ */
+export class OAuthIntegrationData extends WebIntegrationData {
+  // TODO - Validation - Can we get a set of known client ids from config or api call? See https://github.com/mozilla/fxa/pull/15677#discussion_r1291534277
+  @IsOptional()
+  @IsHexadecimal()
+  @bind(T.snakeCase)
+  clientId!: string;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  imageUri: string | undefined;
+
+  @IsOptional()
+  @IsBoolean()
+  @bind()
+  trusted: boolean | undefined;
+
+  @IsOptional()
+  @IsIn(['offline', 'online'])
+  @bind(T.snakeCase)
+  accessType: string | undefined;
```

Comment:
If this is omitted for sync sign in, browser sign in is not complete:
![image](https://github.com/user-attachments/assets/31f52d40-a197-48c3-9c18-e482ad4b1778)

Sync provides a `response_type` param as well, though we don't seem to use it in fxa-settings.

Reply:
I guess same question again, should we add it to our entry point query parameter validation?

Reply:
Concerning `response_type`, is that an oversight on our side... Seems like `auth-client` could take it as option here...

https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/lib/oauth/hooks.tsx#L130

https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-client/lib/client.ts#L2005

Reply:
Probably a good idea to validate on entrypoint in case we're provided with nonsense/invalid values for either of these params? Not sure if we should error of fallback to the defaults though.

``` 
response_type: Joi.string()
  .valid('code')
  .default('code')
  .description(DESCRIPTION.responseType),
```

```
  responseType:
    "Determines the format of the response. 
    Since we only support the authorization-code grant flow, 
    the only permitted value is 'code'.",
```

```
accessType:
    'If specified, a value of `offline` will cause the connecting client 
    to be granted a refresh token alongside its access token.',
```

```
access_type: Joi.string()
  .valid('offline', 'online')
  .default('online')
  .description(DESCRIPTION.accessType),
 ```

Reply:
I'm going to enforce this on query param checks in a quick followup PR.